server/src/main/java/io/deephaven/server/resources/ServerResource.java

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.server.resources;
+
+import java.net.URL;
+
+/**
+ * Represents a set of resources that can be provided by the HTTP server. Priority will be controlled by configuration.
+ */
+public abstract class ServerResource {
+
+    /**
+     * Returns the name of the resource to use in configuration.
+     *
+     * @return the name of the resource
+     */
+    public abstract String getName();
+
+    /**
+     * A URL that can be provided to Jetty to serve the resources.
+     *
+     * @return the URL to the resources
+     */
+    public String getResourceBaseUrl() {
+        // Default implementation assumes that the current class is in the same jar as the resources,
+        // and that there exists a META-INF/resources directory in the jar containing the resources.
+        return getResourceFromClasspath("/META-INF/resources", getClass());
+    }
```

Comment:
Minor: I wonder if we actually want to provide a default impl (or maybe, better as an abstract class, giving it a bit more separation from the interface)? Or, if something besides `/META-INF/resources` would be more appropriatee, given it has special meaning wrt JARs. (We never expect to do `Class.getResource` to access this content...)

Reply:
`META-INF/resources` is a standard in java webapps, not exclusive to jetty or even servlets, and is specifically for "this is static content I want my webserver to be aware of and automatically serve". I'm adding one layer on that, in that some code must opt-in and name that path explicitly, so that we control the ordering of those jars (rather than just scan the classpath).

Not sure what you mean about "We never expect to do `Class.getResource`" here.

Reply:
I may have some misconceptions about `Class.getResource` semantics due to how it works in unit-testing contexts (ie, where there is not typically a JAR for the project you are testing against).